drift-protocol/src/main/java/com/facebook/drift/protocol/TBinaryProtocol.java

```
@@ -391,6 +391,22 @@ public ByteBuffer readBinary()
         return ByteBuffer.wrap(buf);
     }
 
+    @Override
+    public void readBinary(byte[] buf, int offset, int length)
```

Sorry, this interface is really confusing. 

It feels like the interface should be just `byte[] buf, int offset`, and the method should return the length

---

But i think we will need ``` length ``` parameter to call ``` transport.readAll() ``` in ``` ThriftToDriftProtocolReader.java ``` . So is it better to just change method signature to return ``` length ``` instead of ``` void ``` ?

---

You read the length from the wire. The length passed as a parameter makes little sense.

---

As for the `ThriftToDriftProtocolReader`. The (Apache)Thrift Protocol doesn't have similar signature. You can simply throw `UnsupportedOperationException`

---

Okay . got it.